bootstrap/eks/controllers/eksconfig_controller.go

```diff
@@ -35,14 +44,120 @@ type EKSConfigReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+type EKSConfigScope struct {
+	logr.Logger
+	Config      *bootstrapv1.EKSConfig
+	ConfigOwner *bsutil.ConfigOwner
+	Cluster     *clusterv1.Cluster
+}
+
 // +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=eksconfigs,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=eksconfigs/status,verbs=get;update;patch
 
-func (r *EKSConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("eksconfig", req.NamespacedName)
+func (r *EKSConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("eksconfig", req.NamespacedName)
+
+	// get EKSConfig
+	config := &bootstrapv1.EKSConfig{}
+	if err := r.Client.Get(ctx, req.NamespacedName, config); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get config")
+		return ctrl.Result{}, err
+	}
+
+	// check owner references and look up owning Machine object
+	configOwner, err := bsutil.GetConfigOwner(ctx, r.Client, config)
+	if apierrors.IsNotFound(err) {
+		// no error here, requeue until we find an owner
+		return ctrl.Result{}, nil
+	}
+	if err != nil {
+		log.Error(err, "Failed to get owner")
+		return ctrl.Result{}, err
+	}
+	if configOwner == nil {
+		// no error, requeue until we find an owner
+		return ctrl.Result{}, nil
+	}
+
+	log = log.WithValues("kind", configOwner.GetKind(), "version", configOwner.GetResourceVersion(), "name", configOwner.GetName())
+
+	// look up Cluster object associated with owning Machine object
+	cluster, err := util.GetClusterByName(ctx, r.Client, configOwner.GetNamespace(), configOwner.ClusterName())
+	if err != nil {
+		if errors.Cause(err) == util.ErrNoCluster {
+			log.Info(fmt.Sprintf("%s does not belong to a cluster yet, requeueing until it's part of a cluster", configOwner.GetKind()))
+			return ctrl.Result{}, nil
+		}
+		if apierrors.IsNotFound(err) {
+			log.Info("Cluster does not exist yet, requeueing until it is created")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Could not get cluster with metadata")
+		return ctrl.Result{}, err
+	}
+
+	// check for paused annotation
+	if annotations.IsPaused(cluster, config) {
```

Comment:
Do we care if the `configOwner` is paused?

Reply:
Following upstream, I don't think so. I believe the pattern for the pause control is limited to individual objects.